crates/optimism/node/src/engine.rs

```diff
@@ -147,6 +149,143 @@ where
             ))
         }
 
+        if self.chain_spec.is_fork_active_at_timestamp(
+            OptimismHardfork::Holocene,
+            attributes.payload_attributes.timestamp,
+        ) {
+            if attributes.eip_1559_params.is_none() {
+                return Err(EngineObjectValidationError::InvalidParams(
+                    "MissingEip1559ParamsInPayloadAttributes".to_string().into(),
+                ))
+            }
+
+            let (elasticity, denominator) = decode_eip_1559_params(
+                attributes.eip_1559_params.expect("eip_1559_params is none"),
+            );
```

Comment:
we can probably get rid of the expect here like this:
```suggestion
            let Some(eip_1559_params) = attributes.eip_1559_params else {
                return Err(EngineObjectValidationError::InvalidParams(
                    "MissingEip1559ParamsInPayloadAttributes".to_string().into(),
                ))
            };

            let (elasticity, denominator) = decode_eip_1559_params(
                eip_1559_params
            );
```